pennylane/hf/molecule.py

```diff
@@ -0,0 +1,153 @@
+
+        self.symbols = symbols
+        self.coordinates = coordinates
+        self.charge = charge
+        self.mult = mult
+        self.basis_name = basis_name
+
+        self.n_basis, self.basis_data = mol_basis_data(self.basis_name, self.symbols)
+
+        if l is None:
+            l = [i[0] for i in self.basis_data]
+
+        if alpha is None:
+            alpha = [pnp.array(i[1], requires_grad=False) for i in self.basis_data]
+
+        if coeff is None:
+            coeff = [pnp.array(i[2], requires_grad=False) for i in self.basis_data]
+
+        if r is None:
+            r = sum([[self.coordinates[i]] * self.n_basis[i] for i in range(len(self.n_basis))], [])
+
+        self.l = l
+        self.alpha = alpha
+        self.coeff = coeff
+        self.r = r
+
+        self.basis_set = generate_basis_set(self.l, self.alpha, self.coeff, self.r)
+        self.n_orbitals = len(self.l)
+
+        self.nuclear_charges = generate_nuclear_charges(self.symbols)
+
+        self.n_electrons = sum(np.array(self.nuclear_charges)) - self.charge
+        self.core, self.active = qml.qchem.active_space(
+            self.n_electrons,
+            self.n_orbitals,
+            active_electrons=active_electrons,
+            active_orbitals=active_orbitals,
+        )
+
+
+def generate_basis_set(l, alpha, coeff, r):
+    r"""Generate a set of basis function objects.
+
+    Args:
+        l list((tuple[int])): angular momentum numbers of the basis function.
+        alpha list((array(float))): exponents of the Gaussian functions forming basis functions
+        coeff list((array(float))): coefficients of the contracted Gaussian functions
+        r list((array(float))): positions of the Gaussian functions forming the basis functions
+
+    Returns:
+        list(BasisFunction): list containing a set of basis function objects.
+
+    **Example**
+
+    >>> l = [(0, 0, 0), (0, 0, 0)]
+    >>> exponents = [[3.42525091, 0.62391373, 0.1688554], [3.42525091, 0.62391373, 0.1688554]]
+    >>> coefficients = [[0.15432897, 0.53532814, 0.44463454], [0.15432897, 0.53532814, 0.44463454]]
+    >>> centers = [[0.0, 0.0, -0.694349], [0.0, 0.0, 0.694349]]
+    >>>  basis_set = generate_basis_set(l, exponents, coefficients, centers)
+    >>> print(basis_set)
+    [<molecule.BasisFunction object at 0x7f7566db2910>, <molecule.BasisFunction object at 0x7f7566db2a30>]
+    """
+    return [BasisFunction(l[i], alpha[i], coeff[i], r[i]) for i in range(len(l))]
+
+
+def generate_nuclear_charges(symbols):
+    r"""Generate a list of atomic nuclear charges.
+
+    Args:
+    symbols (list[str]): symbols of the atomic species in the molecule
+
+    Returns:
+        list(int): list containing atomic nuclear charges.
+
+    **Example**
+
+    >>> symbols = ["H", "F"]
+    >>> z = generate_nuclear_charges(symbols)
+    >>> print(z)
+    [1, 9]
+    """
+    return [atomic_numbers[s] for s in symbols]
```

Comment:
Same comment here - is the added complexity/overhead from having this function necessary?

Reply:
Removed the function.